vendor/assets/javascripts/behaviors/selectable_asset.js

```diff
@@ -10,6 +10,9 @@ Cmsify.SelectableAsset.prototype.select = function() {
   $preview.html(this.renderAssetOriginalPreview());
   $preview.append($('<p />').html(this.getFileName()));
   $('.js-remove-asset').removeClass('uk-hidden');
+  if (this.$el.data('asset-name') === 'download') {
```

Comment:
Rather than have this conditional, just check if the name field is empty, and set it if it is empty. That way it will be more agnostic to the situation.